### DISCUSSION TITLE
Runtime editor tools?

### DISCUSSION THREAD

**Discussion by jasonswearingen** [TokisanGames/Terrain3D#461] [Questions] [answered]
Hello, I was wondering what suggestions for runtime editing tools/gizmos/plugins are around?

Specifically, I want to allow players to make terrain/building/prop changes in-game, so I need a ui layer to make this work nicely. I am planning on making this myself, but if there are some tips on where I can find similar features, I won't have to recreate the wheel.

Thank you

FYI, I found this, which looks good for placement snapping:  https://github.com/mharitsnf/ExtraSnaps/tree/master

**Comment by TokisanGames** [comment, marked as answer]
Editor.gd is our runtime editor that uses the C++ API. That tool requires collision. Use get_intersection() instead.

**Reply by jasonswearingen** [reply]
thank you, I'll take a look.
